File: verbNet/src/main/java/org/sqlunet/verbnet/style/VerbNetSyntaxSpanner.java
package org.sqlunet.verbnet.style;

import org.sqlunet.style.RegExprSpanner;

/**
 * VerbNet syntax processor
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class VerbNetSyntaxSpanner extends RegExprSpanner
{
	/**
	 * Patterns
	 */
	static private final String[] patterns = { //
			"^([^\\s\n]*)", // cat : 1 capture //
			"^[^\\s\n]* (\\p{Upper}[\\p{Lower}_\\p{Upper}]*)", // value : 1 capture //
	};

	/**
	 * Factories
	 */
	static private final SpanFactory[][] factories = {new SpanFactory[]{VerbNetFactories.catFactory,}, // cat
			new SpanFactory[]{VerbNetFactories.catValueFactory,}, // value
	};

	/**
	 * Constructor
	 */
	public VerbNetSyntaxSpanner()
	{
		super(VerbNetSyntaxSpanner.patterns, VerbNetSyntaxSpanner.factories);
	}
}

File: verbNet/src/main/java/org/sqlunet/verbnet/style/VerbNetSyntaxSpanner.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.verbnet.style

import org.sqlunet.style.RegExprSpanner

/**
 * VerbNet syntax processor
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
object VerbNetSyntaxSpanner : RegExprSpanner(
    arrayOf(
        "^([^\\s\n]*)",  // cat : 1 capture
        "^[^\\s\n]* (\\p{Upper}[\\p{Lower}_\\p{Upper}]*)"
    ),
    arrayOf(
        arrayOf(VerbNetFactories.catFactory),
        arrayOf(VerbNetFactories.catValueFactory)
    )
)
